tensorflow_probability/python/distributions/continuous_bernoulli.py

```diff
@@ -0,0 +1,454 @@
+        if (lams is None) == (logits is None):
+            raise ValueError("Must pass lams or logits, but not both.")
+        self._lims = lims
+        with tf.name_scope(name) as name:
+            self._lams = tensor_util.convert_nonref_to_tensor(
+                lams, dtype_hint=tf.float32, name="lams"
+            )
+            self._logits = tensor_util.convert_nonref_to_tensor(
+                logits, dtype_hint=tf.float32, name="logits"
+            )
+        super(ContinuousBernoulli, self).__init__(
+            dtype=dtype,
+            reparameterization_type=reparameterization.FULLY_REPARAMETERIZED,
+            validate_args=validate_args,
+            allow_nan_stats=allow_nan_stats,
+            parameters=parameters,
+            name=name,
+        )
+
+    @staticmethod
+    def _param_shapes(sample_shape):
+        return {"logits": tf.convert_to_tensor(sample_shape, dtype=tf.int32)}
+
+    @classmethod
+    def _params_event_ndims(cls):
+        return dict(logits=0, lams=0)
+
+    @property
+    def logits(self):
+        """Input argument `logits`."""
+        return self._logits
+
+    @property
+    def lams(self):
+        """Input argument `lams`."""
+        return self._lams
+
+    def _batch_shape_tensor(self):
+        x = self._lams if self._logits is None else self._logits
+        return tf.shape(x)
+
+    def _batch_shape(self):
+        x = self._lams if self._logits is None else self._logits
+        return x.shape
+
+    def _event_shape_tensor(self):
+        return tf.constant([], dtype=tf.int32)
+
+    def _event_shape(self):
+        return tf.TensorShape([])
+
+    def _sample_n(self, n, seed=None):
+        lams = self._lams_parameter_no_checks()
+        cut_lams = tf.where(
+            tf.logical_or(
+                tf.less(lams, self._lims[0]), tf.greater(lams, self._lims[1])
+            ),
+            lams,
+            self._lims[0] * tf.ones_like(lams),
+        )
+        new_shape = tf.concat([[n], tf.shape(lams)], 0)
+        uniform = tf.random.uniform(new_shape, seed=seed, dtype=lams.dtype)
+        sample = tf.where(
+            tf.logical_or(
+                tf.less(lams, self._lims[0]), tf.greater(lams, self._lims[1])
+            ),
+            (
+                tf.math.log(uniform * (2.0 * cut_lams - 1.0) + 1.0 - cut_lams)
+                - tf.math.log(1.0 - cut_lams)
+            )
+            / (tf.math.log(cut_lams) - tf.math.log(1.0 - cut_lams)),
+            uniform,
+        )
+        return tf.cast(sample, self.dtype)
+
+    def _cont_bern_log_norm(self):
+        lams = self._lams_parameter_no_checks()
+        cut_lams = tf.where(
+            tf.logical_or(
+                tf.less(lams, self._lims[0]), tf.greater(lams, self._lims[1])
+            ),
+            lams,
+            self._lims[0] * tf.ones_like(lams),
+        )
+        log_norm = tf.math.log(
+            tf.math.abs(2.0 * tf.math.atanh(1 - 2.0 * cut_lams))
```

Comment:
Replace `tf.math.atanh(1. - 2.0 * cut_lams)` with `0.5 * (tf.math.log1p(-cut_lams) - tf.math.log(cut_lams))`

The latter will be more numerically stable for smaller `cut_lams` and will `inf` far later.